eventuals/grpc/server.cc

```diff
@@ -377,20 +408,53 @@ ServerStatusOrServer ServerBuilder::BuildAndStart() {
 
   builder_.RegisterAsyncGenericService(service.get());
 
-  if (!numberOfCompletionQueues_) {
-    numberOfCompletionQueues_ = 1;
+  if (!number_of_completion_queues_) {
+    if (!completion_thread_pool_) {
+      number_of_completion_queues_ = std::thread::hardware_concurrency();
+    } else {
+      return ServerStatusOrServer{
+          ServerStatus::Error(
+              "if you 'SetCompletionThreadPool()' you must also "
+              "'SetNumberOfCompletionQueues()'"),
+          nullptr};
+    }
   }
 
-  if (!minimumThreadsPerCompletionQueue_) {
-    minimumThreadsPerCompletionQueue_ = 1;
+  if (completion_thread_pool_ && minimum_threads_per_completion_queue_) {
+    return ServerStatusOrServer{
+        ServerStatus::Error(
+            "you can't 'SetCompletionThreadPool()' and "
+            "'SetMinimumThreadsPerCompletionQueue()'"),
+        nullptr};
   }
 
   std::vector<std::unique_ptr<::grpc::ServerCompletionQueue>> cqs;
 
-  for (size_t i = 0; i < numberOfCompletionQueues_.value(); ++i) {
-    cqs.push_back(builder_.AddCompletionQueue());
+  cqs.reserve(number_of_completion_queues_.value());
+
+  for (size_t i = 0; i < number_of_completion_queues_.value(); ++i) {
+    cqs.emplace_back(builder_.AddCompletionQueue());
   }
 
+  auto pool = [&]() -> std::variant<
```

Comment:
Let's add some name for this type since it is used at least twice

Reply:
Great idea! Thank you!